### ISSUE TITLE
Can't call source from submap correctly

### ISSUE THREAD

**Issue by JPDucky** [hyprwm/Hyprland#4485]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit 94d6b2d2c12663efe90f6d0410528b0d7bcc671e dirty (deco: fix missing border on togglefloating (4305)).
Date: Sat Dec 30 09:59:01 2023
Tag: v0.33.1-117-g94d6b2d2
```

</details>

### Bug or Regression?

Bug

### Description

I am trying to do nested submaps while also calling `source` from within said submaps, but it seems that the `source` lines get called at runtime and don't live within the submaps. At first I tried doing nested submaps where I had one submap, "layout", that then had three submaps in it, one for setting dwindle, one for master, and another for hy3- each with their own set of keybinds for each layout. I tried putting the `source` line inside that submap with the thinking being that it would just be called when the submap was entered, but I guess that's not how it works. I also tried to see if I could call the source with a hyprctl and exec but I'm obviously missing something here. 

tl;dr- I'm trying to be able to switch between layouts using keybinds and submaps, as well as loading a set of keybinds with each layout switch, but can't figure it out.

### How to reproduce

Here's the relevant part of my config:
(simplified with just a dwindle example)
```
bind = $mainMod SHIFT, code:59, exec, hyprctl notify 0 3000 0 "submap:layout"
bind = $mainMod SHIFT, code:59, submap, layout
submap = layout

bind = , D, exec, hyprctl --batch "notify 0 3000 0 "submap:dwindle"; keyword general:layout dwindle; dispatch source ./conf/hy3.conf"
>>>>> I tried doing many variations of the following lines without success:
>>>source ./conf/dwindle.conf
>>>exec source ./conf/dwindle.conf
>>>exec source = ./conf/dwindle.conf
bind = , D, submap, reset

# bind = , escape, exec, hyprctl notify 1 3000 0 "submap: return to normal"
bind = , escape, submap, reset
submap = reset
```

**Comment by vaxerski** [comment]
source is evaluated at parse and this cannot be changed. Your option is to dynamically call hyprctl dispatch source ... or modify the files themselves

**Comment by JPDucky** [comment]
`hypdrctl dispatch source` returns "Invalid dispatcher". Is it possible to pass a keybind using `hyprctl`?

**Comment by JPDucky** [comment]
or is it possible to have arguments passed within hyprland.conf?

**Comment by fufexan** [comment]
Try `hyprctl keyword source 'path/to/be/sourced'`.

**Comment by JPDucky** [comment]
That worked! Here's what I ended up with:
```
general {
    gaps_in = 5
    gaps_out = 15
    border_size = 3
    col.active_border = $color11
    col.inactive_border = rgba(ffffffff)
    col.active_border = rgba(33ccffee) rgba(00ff99ee) 45deg
    col.inactive_border = rgba(595959aa)
    layout = dwindle
}
source = ./conf/dwindle.conf


bind = $mainMod SHIFT, code:59, exec, hyprctl notify 0 3000 0 "submap:layout"
bind = $mainMod SHIFT, code:59, submap, layout
submap = layout

bind = , D, exec, hyprctl --batch "notify 0 3000 0 "submap:dwindle"; keyword general:layout dwindle; keyword source ./conf/dwindle.conf"
bind = , D, submap, reset
bind = , M, exec, hyprctl --batch "notify 0 3000 0 "submap:Master"; keyword general:layout master; keyword source ./conf/master.conf"
bind = , M, submap, reset
bind = , N, exec, hyprctl --batch "notify 0 3000 0 "submap:hy3"; keyword general:layout hy3; keyword source ./conf/hy3.conf"
bind = , N, submap, reset

bind = , escape, exec, hyprctl notify 1 3000 0 "submap: cancel"
bind = , escape, submap, reset

submap = reset
```

And that lets me hop around the layouts with their own separate configs. Now if only we could have separate layouts per monitor lol

**Comment by JPDucky** [comment]
just kidding now I've actually tested it and this is what worked:
```
bind = $mainMod SHIFT, code:59, exec, hyprctl notify 0 3000 0 "submap:layout"
bind = $mainMod SHIFT, code:59, submap, layout
submap = layout

bind = , D, exec, hyprctl --batch "notify 0 3000 0 "submap:dwindle"; keyword general:layout dwindle"
bind = , D, exec, hyprctl keyword source ~/dotfiles/hypr/conf/dwindle.conf
bind = , D, submap, reset
bind = , M, exec, hyprctl --batch "notify 0 3000 0 "submap:Master"; keyword general:layout master"
bind = , M, exec, hyprctl keyword source ~/dotfiles/hypr/conf/master.conf
bind = , M, submap, reset
bind = , N, exec, hyprctl --batch "notify 0 3000 0 "submap:hy3"; keyword general:layout hy3"
bind = , N, exec, hyprctl keyword source ~/dotfiles/hypr/conf/hy3.conf
bind = , N, submap, reset

bind = , escape, exec, hyprctl notify 1 3000 0 "submap: cancel"
bind = , escape, submap, reset

submap = reset
```


`hyprctl` didn't want to call the file inside the batch operation so I just added an additional bind to each key.
